pkg/util/psi.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	psiLineFormat = "avg10=%f avg60=%f avg300=%f total=%d"
+
+	ResourceTypeCPU = "cpu"
+	ResourceTypeMem = "mem"
+	ResourceTypeIO  = "io"
+
+	Precision10  = "avg10"
+	Precision60  = "avg60"
+	Precision300 = "avg300"
+
+	DegreeSome = "some"
+	DegreeFull = "full"
+)
+
+type PSILine struct {
+	Avg10  float64
+	Avg60  float64
+	Avg300 float64
+	Total  uint64
+}
+
+type PSIStats struct {
+	Some *PSILine
+	Full *PSILine
+}
+
+type PSIRecord struct {
+	ResourceType string
+	Precision    string
+	Degree       string
+	Value        float64
+}
+
+type PSIByResource struct {
+	CPU PSIStats
+	Mem PSIStats
+	IO  PSIStats
+
+	Records []PSIRecord
```

Comment:
Records []PSIRecord is duplicate in PSIByResource.
these can be generated in metrics pkg when used.